src/mbgl/style/sources/geojson_source_impl.cpp

```diff
@@ -126,20 +144,10 @@ Range<uint8_t> GeoJSONSource::Impl::getZoomRange() {
 std::unique_ptr<Tile> GeoJSONSource::Impl::createTile(const OverscaledTileID& tileID,
                                                       const UpdateParameters& parameters) {
     assert(loaded);
-    if (geoJSONOrSupercluster.is<GeoJSONVTPointer>()) {
-        return std::make_unique<GeoJSONTile>(tileID, base.getID(), parameters,
-            geoJSONOrSupercluster.get<GeoJSONVTPointer>()->getTile(
-                tileID.canonical.z,
-                tileID.canonical.x,
-                tileID.canonical.y).features);
-    } else {
-        assert(geoJSONOrSupercluster.is<SuperclusterPointer>());
-        return std::make_unique<GeoJSONTile>(tileID, base.getID(), parameters,
-            geoJSONOrSupercluster.get<SuperclusterPointer>()->getTile(
-                tileID.canonical.z,
-                tileID.canonical.x,
-                tileID.canonical.y));
-    }
+    std::unique_ptr<Tile> tilePointer = std::make_unique<GeoJSONTile>(tileID, base.getID(), parameters);
```

Comment:
Avoid the need for a `static_cast` here by assigning to a `auto tilePointer = ...`, so that the inferred type is `std::unique_ptr<GeoJSONTile>`. (If the compiler then complains about the `return` statement, make it `return std::move(tilePointer)`.)